include/tscore/MT_hashtable.h

```diff
@@ -267,6 +279,36 @@ IMTHashTable<key_t, data_t>::remove_entry(key_t key)
   return ret;
 }
 
+template <class key_t, class data_t>
+inline data_t
+IMTHashTable<key_t, data_t>::remove_entry(key_t key, data_t data)
```

Comment:
It should not be required to pass in the data, only the key. For non-unique keys, the interface should look like that of `std::multi_map`. That is, when finding by a key, an iterator to the first is returned. To get the range of all keys that are the same, it's `std::multimap::equal_range`. To find  the iterator with specific data, the client walks the range returned. Deleting the element is done by passing the iterator to the `remove` method.